iroh-dns-server/src/main.rs

```diff
@@ -0,0 +1,84 @@
+#![allow(unused_imports)]
+
+use anyhow::Result;
+use axum::{routing::get, Router};
+use clap::Parser;
+use futures::{Future, FutureExt};
+use iroh_dns_server::{self as server, config::Config, dns::DnsServer, state::AppState};
+use iroh_metrics::metrics::start_metrics_server;
+use server::metrics::init_metrics;
+use std::net::{Ipv4Addr, SocketAddr};
+use std::path::PathBuf;
+use tokio::task::JoinSet;
+use tokio_util::sync::CancellationToken;
+use tracing::{debug, debug_span, error, error_span, Instrument, Span};
+
+#[derive(Parser, Debug)]
+struct Cli {
+    /// Path to config file
+    #[clap(short, long)]
+    config: Option<PathBuf>,
+}
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    tracing_subscriber::fmt::init();
+    let args = Cli::parse();
+
+    let config = if let Some(path) = args.config {
+        Config::load(path)?
```

Comment:
this should be async, or in `task::spawn_blocking`

Reply:
:+1: . (but does it matter? it's the first thing that happens, no tasks that could be blocked around yet)

Reply:
makes for less surprises, two refactors later this call happens in a different place and becomes an issue 😅